### PR TITLE
Add/Update/Fix/Run RPC server and client tests

### PR DESCRIPTION
**Pull request by MikeDombo**
*Issue #, if available:*

*Description of changes:*
Makes the existing tests run on build and actually work. 
Adds test for Greengrass client V2.

Fixes Greengrass client V2 which didn't run lambda-based callbacks in the executor.
Makes the socket domain configurable in the v2 client builder so that we can use IP instead of domain sockets.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sdk/pom.xml [hunk 1]
```diff
@@ -45,9 +45,27 @@
       <version>0.15.9</version>
     </dependency>
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <version>4.13.1</version>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.30</version>
+      <scope>compile</scope>
```
Comment (graebm) [inline sdk/pom.xml:51]: `<scope>test</scope>` for slf4j?

**Review by graebm — APPROVED** [review]

File: sdk/pom.xml (hunk 1 above)
Comment (MikeDombo) [reply sdk/pom.xml:51]: I'll try. Verifying.

Comment (MikeDombo) [reply sdk/pom.xml]: Updated

**Comment by MikeDombo** [comment]
Closed for #214.

**Pull request closed without merging**